src/plugins/intel_npu/src/al/include/npu.hpp

```diff
@@ -65,9 +65,7 @@ class IDevice : public std::enable_shared_from_this<IDevice> {
 public:
     using Uuid = ov::device::UUID;
 
-    virtual std::shared_ptr<IExecutor> createExecutor(
-        const std::shared_ptr<const NetworkDescription>& networkDescription,
-        const Config& config) = 0;
+    virtual std::shared_ptr<IExecutor> createExecutor(const std::shared_ptr<IGraph>& graph, const Config& config) = 0;
```

Comment:
IDevice is impacted since we only know IGraph but not networkDescription in abstract layer.